modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -41,7 +41,48 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner,
 
 	k.SetMiddlewareEnabled(ctx, portID, connectionID)
 
-	_, err = k.registerInterchainAccount(ctx, connectionID, portID, version, channeltypes.ORDERED)
+	_, err = k.registerInterchainAccount(ctx, connectionID, portID, version, channeltypes.UNORDERED)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RegisterInterchainAccountWithOrdering is the entry point to registering an interchain account:
+// - It generates a new port identifier using the provided owner string, binds to the port identifier and claims the associated capability.
+// - Callers are expected to provide the appropriate application version string.
+// - For example, this could be an ICS27 encoded metadata type or an ICS29 encoded metadata type with a nested application version.
+// - A new MsgChannelOpenInit is routed through the MsgServiceRouter, executing the OnOpenChanInit callback stack as configured.
+// - An error is returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
+// have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
+//
+// Deprecated: this is a legacy API that is only intended to function correctly in workflows where an underlying authentication application has been set.
```

Comment:
We deprecated this in ibc-go/v6. We should discuss what it would mean to remove these APIs in favour of migrating directly to msg service router, and what users may be affected. 

i.e. removing `RegisterInterchainAccount` public function used by "auth modules" and instead they depend on the baseapp `MsgServiceRouter` and send `MsgRegisterInterchainAccount`.

Reply:
Thanks, @damiannolan. Yes, I agree we should discuss this. I was also thinking that if we get to ICS27 v2 this year, that might be a good moment to tackle that.